### PR TITLE
Rd 2084 prometheus config upgrade 521

### PR DESCRIPTION
**Pull request by glukhman**

### CODE REVIEW

File: cfy_manager/components/prometheus/prometheus.py [hunk 1]
```diff
@@ -201,7 +198,7 @@ def configure(self, upgrade=False):
         self.start()
 
     def upgrade(self):
-        self.configure(upgrade=True)
+        _update_manager_alerts_services()
```
Comment (mateumann) [inline cfy_manager/components/prometheus/prometheus.py:201]:
```suggestion
        _update_manager_alerts_services()
        self.start()
```

File: cfy_manager/components/prometheus/prometheus.py [hunk 2]
```diff
@@ -650,10 +647,6 @@ def _update_manager_alerts_services():
...
 
-    with open(src_file_name) as f:
-        prometheus_conf = f.read()
+    prometheus_conf = files.sudo_read(src_file_name)
     new_services = re.findall(match_pattern, prometheus_conf)[0]
-    with open(dest_file_name) as f:
-        prometheus_conf = f.read()
-    with open(dest_file_name, 'wt') as f:
-        f.write(re.sub(match_pattern, new_services, prometheus_conf))
+    files.replace_in_file(match_pattern, new_services, dest_file_name)
```
Comment (mateumann) [inline cfy_manager/components/prometheus/prometheus.py:652]: `files.replace_in_file()` does not use `sudo` when reading and that's going to cause problems for `managers.yml`.  You'd better use `sudo_read` and then `write_to_file` (which both use sudo).

Comment (glukhman) [reply cfy_manager/components/prometheus/prometheus.py:652]: that's why I change read to sudo_read in replace_in_file below. I think the original was an oversight

File: cfy_manager/components/prometheus/prometheus.py (hunk 1 above)
Comment (glukhman) [reply cfy_manager/components/prometheus/prometheus.py:201]: why though? the upgrade in main.py runs component.start anyway

Comment (mateumann) [reply cfy_manager/components/prometheus/prometheus.py:201]: for the exporters

File: cfy_manager/components/prometheus/prometheus.py (hunk 2 above)
Comment (mateumann) [reply cfy_manager/components/prometheus/prometheus.py:652]: if it was an oversight, let's fix it then, sure

**Review by mateumann — APPROVED** [review]

**Pull request merged**